File: string/basic/pangram-checking/pangram-checking.js
const isPangram = s => {
  // lets remove the numbers if any and change the string to lowercase
  const letters = s.replace(/[^A-Za-z]+/g, "").toLowerCase();
  const letterSet = new Set();
  for (let i = 0; i < letters.length; i++) {
    letterSet.add(letters[i]);
    if (letterSet.size === 26) {
      return true;
      break;
    }
  }
  return false;
};

console.log(isPangram("The quick brown fox jumps over the lazy dog"));
console.log(isPangram("The quick brown fox jumps over the dog"));

File: string/basic/pangram-checking/pangram-checking.py
import re

def is_pangram(s):
    # lets remove the numbers if any and change the string to lowercase
    letters = re.sub('/[^A-Za-z]+/g', '', s)
    letter_set = set()

    for char in letters:
        letter_set.add(char)
        if len(letter_set) == 26:
            return True
    return False

print(is_pangram("The quick brown fox jumps over the lazy dog"))
print(is_pangram("The quick brown fox jumps over the dog"))
